### ISSUE TITLE
Test the parsing of solvers output ?

### ISSUE THREAD

**Issue by tvincent2** [jcavat/rust-lp-modeler#13]
Hello,

I noticed that the parsing of output files generated by solvers is not tested. Shall we add tests for that ?

Best regards,
Thomas

**Comment by jcavat** [comment]
Hello,
Indeed, it would be great to test that!

**Comment by tvincent2** [comment]
Hello,

I will try to add tests for output files for Cbc and GLPK within the next two weeks.
Would it be ok to add a dedicated directory for those files? Something like `src/test-data/` (I'm bad at finding names, better ideas welcome 😃).

Thomas

**Comment by jcavat** [comment]
Hi,
Maybe we should use a test directory called `tests` with:
* one file called `expression.rs` (the current tests)
* one file called `solver.rs` for your purpose

Here is a good resource: https://doc.rust-lang.org/book/second-edition/ch11-03-test-organization.html

What do you think ?

**Comment by tvincent2** [comment]
Hello,

Here is a quick progress report.

- I moved the existing tests into `tests/`. Thanks for the link!
- I had to make `read_solution` functions public in order to test them from the `tests` directory.
- As tests are run in parallel, all of them where trying to write and access to `sol.sol`, which is the default value of `temp_solution_file`, so I added for each solver a function to set a different filename (similarly to what is done in `command_name` functions). However, I am not sure this is the best way as a dummy solver has to be created with `new` just to create the good one with the relevant file name.
- I have some tests for GLPK and Cbc but still need to complete the list to cover all possible cases.

I expect to have more time to work on this by the end of the week. Stay tuned!
